dev/halyard_install_component.sh

```diff
@@ -0,0 +1,83 @@
+#!/bin/bash
+#
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Installs Halyard and installs a single component along with spinnaker-monitoring.
+
+set -e
+
+function show_usage() {
+    fix_defaults
+
+cat <<EOF
+Usage:  $0 [options]
+
+   --component COMPONENT
+       [$COMPONENT]
+       The name of the Spinnaker component to install. Does not contain the 'spinnaker-' prefix.
+
+   --version VERSION
+       [$VERSION]
+       The exact Spinnaker version we are baking images for.
+
+EOF
+}
+
+function process_args() {
+  while [[ $# > 0 ]]; do
+    local key="$1"
+    shift
+
+    case $key in
+        --help)
+            show_usage
+            exit
+            ;;
+
+        --component)
+            COMPONENT=$1
+            shift
+            ;;
+
+        --version)
+            VERSION=$1
+            shift
+            ;;
+
+        *)
+          show_usage
+          >&2 echo "Unrecognized argument '$key'."
+          exit -1
+    esac
+  done
+}
+
+function main() {
+  echo "Installing git..."
+  sudo apt-get install -y git
+
```

Comment:
Previously I needed to copy a subdirectory of Halyard that contained configuration. I'll change this to a curl -> pipe.